src/main/kotlin/org/rust/ide/template/postfix/utils.kt

```diff
@@ -41,3 +42,30 @@ class RustTopMostInScopeSelector(pred: (RustExprElement) -> Boolean) : RustExprP
             .takeWhile { it !is RustBlockElement }
             .any { it is RustExprElement && pred(it) }
 }
+
+class RustAllParentsInScopeSelector(pred: (RustExprElement) -> Boolean) : RustExprParentsSelectorBase(pred) {
+    override fun getExpressions(context: PsiElement, document: Document, offset: Int): List<PsiElement> =
+        context
+            .ancestors
+            .takeWhile { it !is RustBlockElement }
+            .filter { it is RustExprElement && pred(it) }
+            .toList()
+
+    override fun hasExpression(context: PsiElement, copyDocument: Document, newOffset: Int): Boolean =
+        context
+            .ancestors
+            .takeWhile { it !is RustBlockElement }
+            .any { it is RustExprElement && pred(it) }
+}
+
+
+internal fun RustExprElement.isBool() = resolvedType == RustBooleanType
```

Comment:
Unfortunately `internal` modifier is not really useful for us, because the whole project consists of a single module.